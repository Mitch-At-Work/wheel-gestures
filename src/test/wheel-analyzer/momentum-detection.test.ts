import { recordPhases } from '../helper/recordPhases'
import swipeRight from '../fixtures/swipe-right.json'
import swipeRightFast from '../fixtures/swipe-right-fast.json'

describe('momentum detection recognizes', () => {
  it('swipe right', () => {
    expect(recordPhases(swipeRight.wheelEvents)).toMatchSnapshot()
  })

  it('swipe right - fast', () => {
    expect(recordPhases(swipeRightFast.wheelEvents)).toMatchSnapshot()
  })
})
