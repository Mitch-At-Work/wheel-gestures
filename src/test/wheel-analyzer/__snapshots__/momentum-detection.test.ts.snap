// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`momentum detection recognizes swipe right - fast 1`] = `
Object {
  "phases": Array [
    Object {
      "range": Array [
        0,
        27,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "range": Array [
        27,
        96,
      ],
      "wheelType": "MOMENTUM_WHEEL",
    },
    Object {
      "range": Array [
        0,
        96,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe right 1`] = `
Object {
  "phases": Array [
    Object {
      "range": Array [
        0,
        19,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "range": Array [
        19,
        63,
      ],
      "wheelType": "MOMENTUM_WHEEL",
    },
    Object {
      "range": Array [
        0,
        63,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;
